plugins/net_plugin/include/eosio/net_plugin/auto_bp_peering.hpp

```diff
@@ -315,34 +305,41 @@ class bp_connection_manager {
       if (msg.peers.empty())
          return false;
       // initial case, no server_addresses to validate
-      bool initial_msg = msg.peers.size() == 1 && msg.peers[0].server_endpoint.empty();
+      bool initial_msg = msg.peers.size() == 1 && msg.peers[0].bp_peer_info.empty();
       if (!initial_msg) {
          // validate structure and data of msg
-         auto valid_address = [](const std::string& addr) -> bool {
+         auto valid_endpoint = [](const std::string& addr) -> bool {
             const auto& [host, port, type] = net_utils::split_host_port_type(addr);
             return !host.empty() && !port.empty();
```

Comment:
Should we make sure that type is empty?
`return !host.empty() && !port.empty() && type.empty();`

Same at line 143.

Reply:
Probably should since it is the value on their `p2p-server-address` or `p2p-listen-endpoint` that matters.

Reply:
However, 143 is for `p2p-auto-bp-peer` which is outbound and `type` is valid there.